ZenPacks/zenoss/WSMAN/datasources/WSMANDataSource.py

```diff
@@ -224,20 +224,30 @@ def onSuccess(self, results, config):
         result_component_key = ds0.params['result_component_key']
 
         for result in results:
-            if result_component_key:
-                datasource = datasources.get(result[result_component_key])
+            component_key = getattr(result, result_component_key)
+
+            if component_key:
+                datasource = datasources.get(component_key)
 
                 if not datasource:
-                    continue
+                    compatible_datasources = [
```

Comment:
This doesn't seem right to me. Why can't we just make the datasource's result_component_value correct instead of settling for a substring search? If a substring search is a good idea, why do we arbitrarily assume that the first match is the right one?